### PR TITLE
Android package infrastructure

### PR DESCRIPTION
**Pull request by skottmckay**
**Description**: 
Include ORT format model conversion scripts and infrastructure in ORT python package.
  - tweak existing script setup so it can be easily run directly and from the ORT python package
Add config file and readme for Android minimal build package
Update ORT Mobile doco
Disable warning if 'all' optimizations are enabled but NCHWc transformer is excluded (device specific optimizations don't apply in this scenario so the warning is moot).

**Motivation and Context**
Infra for Android package

### CODE REVIEW

File: tools/python/util/convert_onnx_models_to_ort.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+                    models.append(pathlib.Path(root, file))
+
+    if len(models) == 0:
+        raise ValueError("No .onnx files were found in '{}'".format(model_path_or_dir))
+
+    providers = ['CPUExecutionProvider']
+    if use_nnapi:
+        # providers are priority based, so register NNAPI first
+        providers.insert(0, 'NnapiExecutionProvider')
+
```
Comment (gwang-msft) [inline tools/python/util/convert_onnx_models_to_ort.py:75]: The package may not build with NNAPI, can we check if NNAPI is part of get_available_providers, display a warning and move optimization level to 1 here?

File: docs/ONNX_Runtime_for_Mobile_Platforms.md [hunk 2]
```diff
@@ -28,26 +30,29 @@ It is also possible (and optional) to further prune the operator kernel implemen
 This pruning is referred to as "operator type reduction" in this documentation.
 
 - The helper python script requires the standard ONNX Runtime python package to be installed. Install the ONNX Runtime python package from https://pypi.org/project/onnxruntime/. Version 1.5.2 or later is required.
-  To enable operator type reduction, version 1.7 or later is required.
   - `pip install onnxruntime`
   - Ensure that any existing ONNX Runtime python package was uninstalled first, or use `-U` with the above command to upgrade an existing package.
 
-- Additionally, if you want to enable operator type reduction, the Flatbuffers python package should be installed.
+- Additionally, if you want to enable operator type reduction, version 1.7 of ONNX Runtime is required, and the flatbuffers python package must be installed.
```
Comment (gwang-msft) [inline docs/ONNX_Runtime_for_Mobile_Platforms.md:36]: >version 1.7 [](http://example.com/codeflow?start=63&length=11)

version 1.7 or later

File: tools/python/util/convert_onnx_models_to_ort.py (hunk 1 above)
Comment (skottmckay) [reply tools/python/util/convert_onnx_models_to_ort.py:75]: Will add check. 

I think the optimization level is fine though - if someone specifes 'use_nnapi' when running this script we do the special handling to preserve all nodes NNAPI can run.

File: cmake/onnxruntime_python.cmake [hunk 3]
```diff
@@ -281,6 +281,14 @@ file(GLOB onnxruntime_python_datasets_data CONFIGURE_DEPENDS
     "${ONNXRUNTIME_ROOT}/python/datasets/*.onnx"
 )
 
+# Files needed to convert ONNX model to ORT format
+set(onnxruntime_ort_format_model_conversion_srcs
+    ${REPO_ROOT}/tools/python/util/convert_onnx_models_to_ort.py
+    ${REPO_ROOT}/tools/python/util/logger.py
```
Comment (edgchen1) [inline cmake/onnxruntime_python.cmake:287]: maybe we should have some CI test that calls the script from the Python package to verify that we don't miss any dependencies that might be added in the future. that can be in a separate change.

**Review by edgchen1 — APPROVED** [review]

File: cmake/onnxruntime_python.cmake (hunk 3 above)
Comment (skottmckay) [reply cmake/onnxruntime_python.cmake:287]: Will add in separate PR.

**Pull request merged**